src/Symfony/Bridge/Twig/Extension/IntlExtension.php

```diff
@@ -0,0 +1,307 @@
+    }
+
+    /**
+     * Returns a localized language name for a given code.
+     *
+     * @param string      $language
+     * @param string|null $displayLocale
+     */
+    public function getLocalizedLanguage($language, $displayLocale = null)
+    {
+        $separator = false === strpos($language, '-') ? '_' : '-';
+        $params = explode($separator, (string) $language, 2);
+        $lang = $params[0];
+        $region = isset($params[1]) ? $params[1] : null;
+
+        return Intl::getLanguageBundle()->getLanguageName($lang, $region, $displayLocale ?: $this->defaultLocale);
+    }
+
+    /**
+     * Returns a localized locale name for a given code.
+     *
+     * @param string      $locale
+     * @param string|null $displayLocale
+     */
+    public function getLocalizedLocale($locale, $displayLocale = null)
+    {
+        return Intl::getLocaleBundle()->getLocaleName((string) $locale, $displayLocale ?: $this->defaultLocale);
+    }
+
+    /**
+     * Returns a number formatted for the given locale and options.
+     * 
+     * @param mixed  $number
+     * @param string $style
+     * @param string $type
+     * @param null   $displayLocale
+     *
+     * @return bool|string
+     *
+     * @throws \InvalidArgumentException When the type or the style is invalid
+     */
+    public function getLocalizedNumber($number, $style = 'decimal', $type = 'default', $displayLocale = null)
+    {
+        if (empty($this->numberTypeValues)) {
+            $this->numberTypeValues = array(
+                'default' => NumberFormatter::TYPE_DEFAULT,
+                'int32' => NumberFormatter::TYPE_INT32,
+                'int64' => NumberFormatter::TYPE_INT64,
+                'double' => NumberFormatter::TYPE_DOUBLE,
+                'currency' => NumberFormatter::TYPE_CURRENCY,
+            );
+        }
+
+        $typeValues = $this->numberTypeValues;
+        if (!isset($typeValues[$type])) {
+            throw new \InvalidArgumentException(sprintf('The type "%s" does not exist. Known types are: "%s"', $type, implode('", "', array_keys($typeValues))));
+        }
+
+        $formatter = $this->createNumberFormatter(
+            $displayLocale ?: $this->defaultLocale,
+            $style
+        );
+
+        return $formatter->format($number, $typeValues[$type]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'intl_bridge';
+    }
+
+    private function getDefaultTimezone()
+    {
+        if (null === $this->defaultTimezone) {
+            $this->defaultTimezone = date_default_timezone_get();
+        }
+
+        return $this->defaultTimezone;
+    }
+
+    /**
+     * @return IntlDateFormatter
+     *
+     * @throws \InvalidArgumentException On invalid format
+     */
+    private function createDateFormatter($locale, $dateFormat, $timeFormat, $timezone, $calendar, $format)
+    {
+        if ((empty($this->dateFormatValues))) {
```

Comment:
Should be `if (empty($this->dateFormatValues)) {`

This is to avoid autoloading the stub right..? Ie. this could also be the default property value..